tezos/encoding/src/binary_reader.rs

```diff
@@ -500,4 +542,27 @@ mod tests {
         let connection_message_deserialized: ConnectionMessage = de::from_value(&value).unwrap();
         assert_eq!(connection_message, connection_message_deserialized);
     }
+
+    // TODO: add the suggested test for Encoding::Array (also add to both writers)
```

Comment:
@adonagy 
what about this? :)

Reply:
Oh, just a stray TODO, the test below is the one the TODO is mentioning. I'll do a more precise cleanup of the code, once we are satisfied with the functionality.